gh-91565 Simple changes to reflect new bug reporting location.

GadgetSteve opened the pull request:
# gh-91565 Simple changes to reflect new bug reporting location. 

Changes to code & document files to specify the GitHub link for issues rather than the BPO location.
Does not address any changes to Doc/tools/extensions/pyspecific.py #91565

GadgetSteve committed:
Simple changes to reflect new bug reporting location. Does not address any changes to Doc/tools/extensions/pyspecific.py #91565

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

hugovk commented on Doc/README.rst:
```diff
@@ -130,7 +130,7 @@ Contributing
 
 Bugs in the content should be reported to the
-`Python bug tracker <https://bugs.python.org>`_.
+`Python bug tracker <https://github.com/python/cpython/issues>`_.
```
Let's be consistent with [devguide](https://devguide.python.org/tracker/) and call it the "issue tracker":

```suggestion
`Python issue tracker <https://github.com/python/cpython/issues>`_.
```

GadgetSteve replied:
Superseded by later comment

hugovk commented on Doc/README.rst:
```diff
@@ -130,7 +130,7 @@ Contributing
 
 Bugs in the content should be reported to the
-`Python bug tracker <https://bugs.python.org>`_.
+`Python bug tracker <https://github.com/python/cpython/issues>`_.
```
And in fact, shall we link to https://devguide.python.org/tracker/ instead? 

That has info about checking issues don't already exist, might help avoid duplicates.

```suggestion
`Python issue tracker <https://devguide.python.org/tracker/>`_.
```

GadgetSteve replied:
Much better!

hugovk commented on Lib/locale.py:
```diff
@@ -1475,7 +1475,7 @@ def getpreferredencoding(do_setlocale=True):
 #
 # NOTE: this mapping is incomplete.  If your language is missing, please
-# submit a bug report to the Python bug tracker at http://bugs.python.org/
+# submit a bug report to the Python bug tracker at https://github.com/python/cpython/issues
```
```suggestion
# submit a bug report to the Python bug tracker: https://devguide.python.org/tracker/
```

hugovk commented on Lib/platform.py:
```diff
@@ -10,7 +10,8 @@
 #    This module is maintained by Marc-Andre Lemburg <mal@egenix.com>.
 #    If you find problems, please submit bug reports/patches via the
-#    Python bug tracker (http://bugs.python.org) and assign them to "lemburg".
+#    Python bug tracker (https://github.com/python/cpython/issues) and
```
```suggestion
#    Python bug tracker (https://devguide.python.org/tracker/) and
```

ezio-melotti replied:
This was marked as solved, but the suggestion hasn't been committed.  Not sure if it was intentional or an oversight.

I see two options here.  If we want to keep a direct link to the issue tracker, we can just tweak the original sentence
```suggestion
#    Python issue tracker (https://github.com/python/cpython/issues) and
```

If we want to link to the devguide as @hugovk was suggesting, the comment should be rephrased a bit and the devguide URL updated.

encukou replied:
For the target audience here, I don't think the devguide adds much. The main info is who to tag.

hugovk commented on Python/dtoa.c:
```diff
@@ -72,7 +72,7 @@
  * at acm dot org, with " at " changed at "@" and " dot " changed to ".").
  * Please report bugs for this modified version using the Python issue tracker
- * (http://bugs.python.org). */
+ * (https://github.com/python/cpython/issues). */
```
```suggestion
 * (https://devguide.python.org/tracker/). */
```

hugovk commented on Mac/README.rst:
```diff
@@ -334,9 +334,9 @@ The configure script sometimes emits warnings like the one below::
-   configure: WARNING:     ## Report this to https://bugs.python.org/ ##
-   configure: WARNING:     ## --------------------------------------- ##
+   configure: WARNING:     ## ------------------------------------------------------- ##
+   configure: WARNING:     ## Report this to https://github.com/python/cpython/issues ##
```
```suggestion
   configure: WARNING:     ## Report this to https://devguide.python.org/tracker/ ##
```

hugovk reviewed (commented):
Thanks, a few suggestions.

GadgetSteve committed:
Update Doc/README.rst

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

GadgetSteve committed:
Update Lib/platform.py

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

GadgetSteve committed:
Update Mac/README.rst

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

GadgetSteve committed:
Update Python/dtoa.c

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

GadgetSteve committed:
Update Lib/locale.py

Co-authored-by: Hugo van Kemenade <hugovk@users.noreply.github.com>

arhadthedev committed:
Update configure.ac

arhadthedev committed:
Regenerate configure

GadgetSteve committed:
Merge pull request #6 from arhadthedev/issue_91565

Amendment for "Simple changes to reflect new bug reporting location"

GadgetSteve committed:
Reword links to issues to make it clearer they are to the devguide. gh-91565

GadgetSteve committed:
Wording changes to configure overidden by autoconf. gh-91565

ronaldoussoren commented on Lib/platform.py:
```diff
@@ -9,8 +9,9 @@
-#    Python bug tracker (http://bugs.python.org) and assign them to "lemburg".
+#    If you find problems, please submit bug reports/patches as detailed
+#    in the Python devguide (https://devguide.python.org/tracker/) and
+#    assign them to "lemburg".
```
"lemburg" should be "malemburg" here.

GadgetSteve replied:
@ronaldoussoren Thanks for pointing that out correction being pushed in a few minutes.

ronaldoussoren reviewed (commented):
I don't have anything to add other than a minor change to Lib/platform.py due to Marc-Andre's GH user name being different than his Roundup user name.

The change to Mac/README.rst looks good to me, as do the other changes.

GadgetSteve committed:
Corrected lemburg to malemburg in Lib/platform.py. gh-91565

GadgetSteve committed:
Changes to configure.ac to correct wording of bug tracker reference in header. gh-91565

GadgetSteve committed:
Afer running autoconf, configure then make regen-all

encukou committed:
Merge in the main branch

encukou committed:
Adjust links

- Devguide URL is now triage/issue-tracker/
- That page is for bug reports only; for "bugs/patches" use GitHub
  directly
- Outside contributors can't assign issues; they need to mention the
  username instead.
- For crashers, let's use "issue number/link", the target audience
  here should know we use GitHub.

GadgetSteve committed:
Merge branch 'main' into issue_91565

encukou committed:
Update Lib/platform.py

Co-authored-by: Ezio Melotti <ezio.melotti@gmail.com>
